src/checkers/inference/InferenceAnnotatedTypeFactory.java

```diff
@@ -166,6 +166,14 @@ public CFTransfer createFlowTransferFunction(CFAbstractAnalysis<CFValue, CFStore
 
     @Override
     public TreeAnnotator createTreeAnnotator() {
+        if (realTypeFactory instanceof InferrableAnnotatedTypeFactory) {
+            TreeAnnotator realInferenceTreeAnnotator = ((InferrableAnnotatedTypeFactory) realTypeFactory)
+                    .getRealInferenceTreeAnnotator(this, realChecker,
+                            realTypeFactory, variableAnnotator, slotManager);
+            return new ListTreeAnnotator(new ImplicitsTreeAnnotator(this),
```

Comment:
I would just return what the realTypeFactory returns. Make it the obligation of that method to create a ListTreeAnnotator - if it so chooses.